src/AutoRest.CSharp.V3/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -57,7 +57,14 @@ public static void WriteParameterNullChecks(this CodeWriter writer, IReadOnlyCol
             {
                 if (parameter.DefaultValue != null && !CanBeInitializedInline(parameter))
                 {
-                    writer.Line($"{parameter.Name} ??= new {parameter.Type}({parameter.DefaultValue.Value.Value:L});");
+                    if (TypeFactory.IsStruct(parameter.Type))
+                    {
+                        writer.Line($"{parameter.Name} = {parameter.Name} ?? {parameter.DefaultValue.Value.Value:L};");
```

Comment:
Let's use the same ` {parameter.Name} ??= ` syntax for both.

Can we extract the second half and move it to `WriteConstant` method too?